### PR TITLE
CLOUDP-68204: Load default mongodb major version from Atlas

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-68204

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

Default mongodb version for atlas will be dynamically loaded

### CODE REVIEW

File: internal/cli/atlas/clusters/create.go [hunk 1]
```diff
@@ -227,6 +226,10 @@ func CreateBuilder() *cobra.Command {
 			return opts.Run()
 		},
 	}
+	currentMDBVersion, err := DefaultMongoDBMajorVersion()
+	if err != nil {
+		panic(err)
```
Comment (gssbzn) [inline internal/cli/atlas/clusters/create.go:231]: TODO: improve this

File: internal/cli/opsmanager/versionmanifest/update.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ func (opts *UpdateOpts) initStore() error {
 	if err != nil {
 		return err
 	}
-	opts.storeStaticPath, err = store.NewStaticPath(config.Default())
+	opts.storeStaticPath, err = store.NewForVersionManifest(config.Default())
```
Comment (threebee) [inline internal/cli/opsmanager/versionmanifest/update.go:44]: [nit - feel free to ignore] Could we rename `NewForVersionManifest` method something like `NewVersionManifestPath`? `NewForVersionManifest ` method name is confusing to me.

**Review by threebee — APPROVED** [review]
LGTM, I have left a small nit.

File: internal/cli/atlas/clusters/create.go [hunk 3]
```diff
@@ -30,14 +30,13 @@ import (
 )
```
Comment (andreaangiolillo) [inline internal/cli/atlas/clusters/create.go:30]:
```suggestion
// Copyright 2021 MongoDB Inc
```

File: internal/cli/atlas/clusters/create_test.go [hunk 4]
```diff
@@ -42,7 +42,7 @@ func TestCreateOpts_Run(t *testing.T) {
 			members:    3,
```
Comment (andreaangiolillo) [inline internal/cli/atlas/clusters/create_test.go:42]: The Copyright is 2020

File: internal/store/store.go [hunk 5]
```diff
@@ -223,6 +224,31 @@ func NewStaticPath(c Config) (*Store, error) {
 	return s, nil
 }
 
+// NewPrivate get the appropriate client for the profile/service selected
+func NewForPrivateUnauth(c Config) (*Store, error) {
```
Comment (andreaangiolillo) [inline internal/store/store.go:228]:
```suggestion
// NewForPrivateUnauth gets the appropriate client for the profile/service selected
func NewForPrivateUnauth(c Config) (*Store, error) {
```

File: internal/store/store.go [hunk 6]
```diff
@@ -223,6 +224,31 @@ func NewStaticPath(c Config) (*Store, error) {
...
+	}
+
+	client, err := defaultClient(c)
+	if err != nil {
+		return nil, err
+	}
+	err = s.setAtlasClient(client)
+	if err != nil {
+		return nil, err
+	}
```
Comment (andreaangiolillo) [inline internal/store/store.go:247]:
```suggestion
	if err = s.setAtlasClient(client); err != nil {
	     return nil, err
	}
```

File: internal/cli/atlas/clusters/create.go [hunk 7]
```diff
@@ -246,3 +247,17 @@ func CreateBuilder() *cobra.Command {
...
+func DefaultMongoDBMajorVersion() (string, error) {
+	if defaultMongoDBMajorVersion != "" {
+		return defaultMongoDBMajorVersion, nil
+	}
+	s, err := store.NewForPrivateUnauth(config.Default())
+	if err != nil {
+		return "", err
+	}
+	defaultMongoDBMajorVersion, err = s.DefaultMongoDBVersion()
+	return defaultMongoDBMajorVersion, err
```
Comment (andreaangiolillo) [inline internal/cli/atlas/clusters/create.go:262]:
```suggestion
	return s.DefaultMongoDBVersion()
```

**Review by andreaangiolillo — CHANGES_REQUESTED** [review]
Great improvement! Left few suggestions

File: internal/cli/atlas/clusters/create.go (hunk 3 above)
Comment (gssbzn) [reply internal/cli/atlas/clusters/create.go:30]: We don't need to change the year as we update files

File: internal/cli/opsmanager/versionmanifest/update.go (hunk 2 above)
Comment (gssbzn) [reply internal/cli/opsmanager/versionmanifest/update.go]: yep, I don't like it either, I've been wanting to do a refactor on how we set up the store but I'll leave that for another time

File: internal/cli/atlas/clusters/create.go (hunk 3 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/clusters/create.go:30]: ah okay! Thanks!

**Review by andreaangiolillo — DISMISSED** [review]
LGTM

**Comment by gssbzn** [comment]
evergreen retry

**Review by threebee — DISMISSED** [review]
LGTM!

**Comment by andreaangiolillo** [comment]
Cluster failures seem legit https://evergreen.mongodb.com/test_log/mongocli_master_e2e_atlas_clusters_atlas_clusters_e2e_patch_ed96f83cc367158a4987431c378608ba06d14e96_600ab19fe3c33131ed22b6a4_21_01_22_11_06_08/0/github.com_mongodb_mongocli_e2e_atlas.TestClustersFile_Create_via_file. I think you have to update the e2e test as well

**Comment by gssbzn** [comment]
@andreaangiolillo we actually had a bug in our e2e tests

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Comment by gssbzn** [comment]
Failures are on datalake, we may need to update some things there

**Comment by andreaangiolillo** [comment]
I have created https://jira.mongodb.org/browse/CLOUDP-80931 to fix it

**Pull request merged**